presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java

```diff
@@ -537,6 +541,22 @@ private JoinEnumerationResult setJoinNodeProperties(JoinNode joinNode)
             if (isAtMostScalar(joinNode.getLeft(), lookup)) {
                 return createJoinEnumerationResult(joinNode.flipChildren().withDistributionType(REPLICATED));
             }
+
+            if (isBelowMaxBroadcastSize(joinNode, context) && isBelowMaxBroadcastSize(joinNode.flipChildren(), context)) {
```

Comment:
It would be better to call into DetermineJoinDistributionType here so as not to repeat code.  Also, that would enable us to incorporate other features of DetermineJoinDistributionType (e.g. size based join).